.github/workflows/ci-llama-tests.yaml

```diff
@@ -0,0 +1,79 @@
+name: "Llama Benchmarking Tests"
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches:
+      - main
+  schedule:
+    - cron: "0 11 * * 1-5"  # Weekdays at 11:00 AM UTC
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.number || github.sha }}
+  cancel-in-progress: true
+
+jobs:
+  test_llama_quick:
+    name: "Llama Benchmarking 8B Quick Tests"
+    strategy:
+      matrix:
+        version: [3.11]
+      fail-fast: false
+    runs-on: linux-mi300-1gpu-ossci
+    steps:
+      - uses: actions/checkout@v4.2.2
+
+      - name: Setup Python and Install Dependencies
+        uses: ./.github/actions/test-setup
+        with:
+          python-version: ${{ matrix.version }}
+
+      - name: Run Quick Llama 8B Test
+        run: |
+          source ${{ github.workspace }}/.venv/bin/activate
+          pytest sharktank/tests/models/llama/benchmark_amdgpu_test.py -v -s \
+            --iree-hip-target=gfx942 --iree-device=hip://0 --run-quick-llama-test
+
+      - name: Upload Llama Executable Files
+        uses: actions/upload-artifact@v4.6.0
+        with:
+          name: llama-files
+          path: ${{ github.workspace }}/${{ env.date }}
+
+  test_llama_large:
+    if: ${{ github.repository_owner == 'nod-ai' || github.event_name != 'schedule' }}
+    name: "Llama Benchmarking Full Tests"
+    strategy:
+      matrix:
+        version: [3.11]
+      fail-fast: false
+    runs-on: linux-mi300-1gpu-ossci
+    steps:
+      - uses: actions/checkout@v4.2.2
+
+      - name: Setup Python and Install Dependencies
+        uses: ./.github/actions/test-setup
+        with:
+          python-version: ${{ matrix.version }}
+
+      - name: Run Full Llama Benchmarking Tests
+        run: |
+          source ${{ github.workspace }}/.venv/bin/activate
+          pytest sharktank/tests/models/llama/benchmark_amdgpu_test.py -v -s \
+            --run-nightly-llama-tests --iree-hip-target=gfx942 --iree-device=hip://0 \
+            --html=out/llm/llama/benchmark/index.html
+
+      - name: Deploy to GitHub Pages
+        uses: peaceiris/actions-gh-pages@v4.0.0
+        with:
+          github_token: ${{ secrets.SHARK_PLATFORM_GH_TOKEN }}
```

Comment:
As you don't have access to the secret from a fork your workflow will fail. Maybe comment this out and enable if / after landing.